server/api/itemRouter.js

```diff
@@ -0,0 +1,115 @@
+const Item = require('../db/models/item');
+const Category = require('../db/models/category');
+const express = require('express');
+const router = express.Router();
+
+/**
+ * Add an item
+ * 
+ * req.body: {sourceLink: String,
+ *            highlight: String,
+ *            userId: mongoose.ObjectId,
+ *            placesId: id(optional),
+ *            comment: String (optional),}
+ * 
+ * response: {success: boolean,
+ *            message: String}
+ */
+router.post('/addItem', (req, res, next) => {
+  const newItem = req.body;
+  const item = new Item(newItem);
+  let response = {}; 
+  item.save((err) => {
+    if (err) {
+      response.success = false;
+      response.message = 'error adding item';
+      res.json(response);
+      next(err);
+    } else {
+      response.success = true;
+      response.message = 'item added';
+      res.json(response);
+    }
+  });
+});
+
+/**
+ * Get all items
+ * 
+ * req.body: {userId: mongoose.objectId}
+ * 
+ * response: [{sourceLink: String,
+ *            highlight: String,
+ *            placeId: id(optional),
+ *            comments: String (optional),}]
+ */
+router.get('/getItems', (req, res, next) => {
+  Item.find({ userId: req.body.userId }, (err, items) => {
+    if (err) {
+      next(err);
+    }
+    if (items) {
+      res.json(items);
+    }
+  });
+});
+
+/**
+ * Add item to category
+ * 
+ * req.body: {itemId: mongoose.objectId,
+ *            categoryId: mongoose.objectId}
+ * 
+ * response: {success: boolean,
+ *            message: String}
+ */
+router.put('/addItemToCategory', async (req, res, next) => {
+  let response = {}; 
+  try {
+    const itemId = req.body.itemId;
+    const categoryId = req.body.categoryId;
+    await Item.update({ _id: itemId }, { $push: { categoryIds: this.categoryId } }, done).exec();
+    await Category.update({ _id: categoryId }, { $push: { items: this.itemId } }, done).exec();
+    response.success = true;
```

Comment:
Since we aren't doing anything special with this response here, I think it'd be better to use the res "status" and "send" methods.

Reply:
okay will correct!